code/modules/logging/log_entry.dm

```diff
@@ -0,0 +1,114 @@
+
+// Schema version must always be the very last element in the array.
+
+// Current Schema: 1.0.0
+// [timestamp, category, message, data, world_state, semver_store, id, schema_version]
+
+/// A datum which contains log information.
+/datum/log_entry
+	/// Next id to assign to a log entry.
+	var/static/next_id = 0
+
+	/// Unique id of the log entry.
+	var/id
+
+	/// Schema version of the log entry.
+	var/schema_version = "1.0.0"
+
+	/// Unix timestamp of the log entry.
+	var/timestamp
+
+	/// Category of the log entry.
+	var/category
+
+	/// Message of the log entry.
+	var/message
+
+	/// Data of the log entry; optional.
+	var/list/data
+
+	/// Semver store of the log entry, used to store the schema of data entries
```

Comment:
Could be nice to have an explanation somewhere of both what a Semver is, and what a Semver Store is. Because even I had to Google it, after I couldn't find an explanation on here.